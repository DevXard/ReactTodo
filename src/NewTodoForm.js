import {useState} from 'react';

const NewTodoForm = ({addNewTodo}) => {
    const [formData, setFormData] = useState('')

    const handleChange = (e) => {
        e.preventDefault()
        setFormData(e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault()
        addNewTodo(formData)
    }
    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="todo">Todo</label>
            <input 
            id="todo"
            type="text"
            name='todo'
            value={formData}
            onChange={handleChange}
            />
            <button>Add Todo</button>
        </form>
    )
}

export default NewTodoForm;