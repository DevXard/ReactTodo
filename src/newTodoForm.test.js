import { render, fireEvent } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

it("Should render without crasshing", () => {
    render(<NewTodoForm />)
})

it("Matches snapshot", () => {
    const{asFragment} = render(<NewTodoForm />)
    expect(asFragment()).toMatchSnapshot()
})