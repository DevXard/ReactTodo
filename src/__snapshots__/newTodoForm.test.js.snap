// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="todo"
    >
      Todo
    </label>
    <input
      id="todo"
      name="todo"
      type="text"
      value=""
    />
    <button>
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;
